private_gpt/server/chat/chat_service.py

```diff
@@ -97,8 +98,11 @@ def _chat_engine(
         system_prompt: str | None = None,
         use_context: bool = False,
         context_filter: ContextFilter | None = None,
+        context_template: str | None = None,
     ) -> BaseChatEngine:
         if use_context:
+            if context_template is None:
```

Comment:
Instead of this, why not checking if settings.rag.default_cotnext_template is not None and then apply it?